### ISSUE TITLE
Target controller remove finalizer after timeout could leave orphan deployment without symphony objects

### ISSUE THREAD

**Issue by Haishi2016** [eclipse-symphony/symphony#19]
(reported by @RemindD)

[Target controller](https://github.com/Azure/symphony/blob/main/k8s/controllers/fabric/target_controller.go) calls removeFinalizer when timeout waiting for summary status.
This can cause some issues when job manager picks up reconcile job very slowly (for example, job manager has some logic to delay or skip job). If job manager picks up the job after target controller already removed finalizer, then the symphony object is also deleted which causes job manager unable to get targets in the state provider. And deployment associated with this target is also left.
Potential fix is to return delete failure rather than remove finalizer silently.

**Comment by Haishi2016** [comment]
It's hard to detect if an error is un-recoverable. We have to make sure we don't get an object infinitely stuck.

**Comment by msftcoderdjw** [comment]
@RemindD , after we introducing new target controller implemented by AIO team, please evaluate if this issue still persists?

**Comment by RemindD** [comment]
@msftcoderdjw yes. The issue still exists but the timeout has been changed from 5 mins to 15 mins

**Comment by msftcoderdjw** [comment]
Assign @RemindD to evaluate.

**Comment by msftcoderdjw** [comment]
1. Mark delete fail or remove object directly? - remove CR or uninstall chart
2. Make sure the on-going solution-manager reconcile to be finished? (reconcile process will depend on k8s object or not?)
3. Orphan deployment summary/state in timeout.
